### PR TITLE
Add more detail to DCD timestep deprecation message

### LINKED ISSUE #3923: Improve deprecation warning for DCDReader copying timestep
Issue for proper paper trail

Add a deprecation warning for the DCDReader copying timestep unlike the other readers. Behaviour documented in #3889.

paraphrasing:

### Expected behavior ###

The readers should all have the same behaviour with respect to iterating through the trajectory and changing the base values in the `position` attribute. 

### Actual behavior ###

The DCD reader creates a new **copy** of the current timestep which allows the coordinates in each frame to vary independently.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #3923 

Changes made in this Pull Request:
 - Adds more detail to deprecation warning. 

PR Checklist
------------
 - [X] Tests?
 - [X] Docs?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

**Comment by hmacdope** [comment]
Related #3889

File: package/MDAnalysis/coordinates/DCD.py [hunk 1]
```diff
@@ -164,8 +164,14 @@ def __init__(self, filename, convert_units=True, dt=None, **kwargs):
...
         warnings.warn("DCDReader currently makes independent timesteps"
                       " by copying self.ts while other readers update"
-                      " self.ts inplace. This behaviour will be changed in"
-                      " 3.0 to be the same as other readers",
+                      " self.ts inplace. This behavior will be changed in"
+                      " 3.0 to be the same as other readers. In practice this"
+                      " means that you will no longer be able to store independent"
+                      " references to the Timestep object associated with the"
+                      " DCDReader. Read more at"
+                      " https://github.com/MDAnalysis/mdanalysis/issues/3899"
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/DCD.py]: This is about *Bump up codecov action, add token secret* — probably wrong URL.

**Review by orbeckst — CHANGES_REQUESTED** [review]
Sorry, quick review. 

As a user I'd still not understand what's changing. I'd want to see an example that demonstrates the code pattern that will fail and how to change it. Either add and example to #3923 or create an entry in our [WIki FAQ](https://github.com/MDAnalysis/mdanalysis/wiki/FAQ) (the [User Guide FAQ](https://userguide.mdanalysis.org/stable/faq.html) would also be a reasonable place).

File: package/MDAnalysis/coordinates/DCD.py (hunk 1 above)
Comment (hmacdope) [reply package/MDAnalysis/coordinates/DCD.py]: Whoops sorry.

File: package/MDAnalysis/coordinates/DCD.py [hunk 2]
```diff
@@ -164,8 +164,14 @@ def __init__(self, filename, convert_units=True, dt=None, **kwargs):
         self.ts.dt = dt
         warnings.warn("DCDReader currently makes independent timesteps"
                       " by copying self.ts while other readers update"
-                      " self.ts inplace. This behaviour will be changed in"
-                      " 3.0 to be the same as other readers",
+                      " self.ts inplace. This behavior will be changed in"
+                      " 3.0 to be the same as other readers. In practice this"
+                      " means that you will no longer be able to store independent"
+                      " references to the Timestep object associated with the"
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/DCD.py]: I'm going to +1 @orbeckst here, "references to Timestep" is _a lot_ for most users. I'd suggest probably putting up some code examples of how the behaviour is going to change.

File: package/MDAnalysis/coordinates/DCD.py [hunk 3]
```diff
@@ -164,8 +164,14 @@ def __init__(self, filename, convert_units=True, dt=None, **kwargs):
...
+                      " DCDReader. Read more at"
+                      " https://github.com/MDAnalysis/mdanalysis/issues/3899"
+                      " and https://github.com/MDAnalysis/mdanalysis/issues/3923"
+                      " to learn if this change in behavior might affect you.",
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/DCD.py]:
```suggestion
                      " self.ts inplace. This behavior will be changed in"
                      " 3.0 to be the same as other readers. Read more at"
                      " https://github.com/MDAnalysis/mdanalysis/issues/3889"
                      " to learn if this change in behavior might affect you.",
```
For the sake of not making this take too long, could we be happy with this and we can workshop the exact details on what needs to be communicated there?

**Review by IAlibay — CHANGES_REQUESTED** [review]

File: package/MDAnalysis/coordinates/DCD.py (hunk 3 above)
Comment (orbeckst) [reply package/MDAnalysis/coordinates/DCD.py]: sounds good to me

**Review by IAlibay — APPROVED** [review]

**Comment by hmacdope** [comment]
👍 Cheers @IAlibay

**Review by orbeckst — APPROVED** [review]

**Pull request merged**